Enhance release creation with app dropping and improved UX

njuettner opened the pull request:
Add app dropping support and improve bumpall UX

### Checklist

- [x] Update changelog in CHANGELOG.md.

Gacko reviewed (approved):
LGTM!

Just one idea, not related to this PR, but when I was using `devctl` to create a Release recently, I was hoping for a way to also define dependencies, maybe as a comma-separated list in the `<app>@<version>` string. So if you got some time, you could also add something like that. ❤️

njuettner commented:
@Gacko Good idea I added this now, was a quick win 🙂 

I added a description how to do that now dropping, adding or overwriting dependencies:

```
devctl release create -h
Creates and registers a new Giant Swarm platform release including Release CR and release notes.

Usage:
  devctl release create [flags]

Examples:
  # Create a new release by manually specifying new app and component versions
  devctl release create \
        --name v30.1.0 \
        --provider capa \
        --base v30.0.0 \
        --app cert-manager@3.9.2 \
        --component cluster-aws@4.0.1 \
        --overwrite

  # Create a release by bumping all apps and components to their latest versions
  devctl release create \
        --name v30.1.0 \
        --provider capa \
        --base v30.0.0 \
        --bumpall

  # Bump all components, but override a specific app's version and add a dependency
  # This will overwrite any existing dependencies on 'coredns'.
  devctl release create \
        --name v30.1.0 \
        --provider capa \
        --base v30.0.0 \
        --bumpall \
        --app coredns@1.2.3@@cilium

  # Bump all components and remove all dependencies from a specific app
  # This will drop any existing dependencies on 'coredns'.
  devctl release create \
        --name v30.1.0 \
        --provider capa \
        --base v30.0.0 \
        --bumpall \
        --app coredns@1.2.3@@
```

I also added it into the UX:

<img width="964" height="779" alt="image" src="https://github.com/user-attachments/assets/7f6c25c6-0504-4858-a9d0-b1331183fe51" />

njuettner commented:
I updated it again for dependencies colors in case we add a remove a dependency otherwise it will not present in green (added) or red (removed)

<img width="965" height="779" alt="image" src="https://github.com/user-attachments/assets/016a695c-c5ab-481b-ae87-63cc0d3086db" />

In case we add something:

Example `devctl release create --base v31.1.1 --name v32.0.0 --provider vsphere --bumpall --app cilium@1.2.3@@new-dependency  --overwrite`

<img width="1012" height="780" alt="image" src="https://github.com/user-attachments/assets/209d36f4-d23b-48ea-97ff-c575ec7fa64c" />

Gacko commented:
I was first wondering what's the double `@` for, but then saw you can also already define the component version. Makes sense to me! I think one could discuss about what separator to use for what, like maybe using a format like `app@version(component-version)#dependency,dependency`, but that's nit picking.
